bentoml/grpc/utils/__init__.py

```diff
@@ -0,0 +1,177 @@
+        HTTPStatus.TOO_MANY_REQUESTS: grpc.StatusCode.UNAVAILABLE,
+        HTTPStatus.BAD_GATEWAY: grpc.StatusCode.UNAVAILABLE,
+        HTTPStatus.SERVICE_UNAVAILABLE: grpc.StatusCode.UNAVAILABLE,
+        HTTPStatus.GATEWAY_TIMEOUT: grpc.StatusCode.DEADLINE_EXCEEDED,
+        HTTPStatus.BAD_REQUEST: grpc.StatusCode.INVALID_ARGUMENT,
+        HTTPStatus.INTERNAL_SERVER_ERROR: grpc.StatusCode.INTERNAL,
+        HTTPStatus.UNPROCESSABLE_ENTITY: grpc.StatusCode.FAILED_PRECONDITION,
+    }
+
+
+@lru_cache(maxsize=1)
+def grpc_status_to_http_status_map() -> dict[grpc.StatusCode, Enum]:
+    return {v: k for k, v in http_status_to_grpc_status_map().items()}
+
+
+@lru_cache(maxsize=1)
+def filetype_pb_to_mimetype_map() -> dict[pb.File.FileType.ValueType, str]:
+    return {
+        pb.File.FILE_TYPE_CSV: "text/csv",
+        pb.File.FILE_TYPE_PLAINTEXT: "text/plain",
+        pb.File.FILE_TYPE_JSON: "application/json",
+        pb.File.FILE_TYPE_BYTES: "application/octet-stream",
+        pb.File.FILE_TYPE_PDF: "application/pdf",
+        pb.File.FILE_TYPE_PNG: "image/png",
+        pb.File.FILE_TYPE_JPEG: "image/jpeg",
+        pb.File.FILE_TYPE_GIF: "image/gif",
+        pb.File.FILE_TYPE_TIFF: "image/tiff",
+        pb.File.FILE_TYPE_BMP: "image/bmp",
+        pb.File.FILE_TYPE_WEBP: "image/webp",
+        pb.File.FILE_TYPE_SVG: "image/svg+xml",
+    }
+
+
+@lru_cache(maxsize=1)
+def mimetype_to_filetype_pb_map() -> dict[str, pb.File.FileType.ValueType]:
+    return {v: k for k, v in filetype_pb_to_mimetype_map().items()}
+
+
+def grpc_status_code(err: BentoMLException) -> grpc.StatusCode:
+    """
+    Convert BentoMLException.error_code to grpc.StatusCode.
+    """
+    return http_status_to_grpc_status_map().get(err.error_code, grpc.StatusCode.UNKNOWN)
+
+
+def to_http_status(status_code: grpc.StatusCode) -> int:
+    """
+    Convert grpc.StatusCode to HTTPStatus.
+    """
+    status = grpc_status_to_http_status_map().get(
+        status_code, HTTPStatus.INTERNAL_SERVER_ERROR
+    )
+
+    return status.value
+
+
+@dataclass
+class MethodName:
+    """
+    Represents a gRPC method name.
+
+    Attributes:
+        package: This is defined by `package foo.bar`, designation in the protocol buffer definition
+        service: service name in protocol buffer definition (eg: service SearchService { ... })
+        method: method name
+    """
+
+    package: str = ""
+    service: str = ""
+    method: str = ""
+
+    @property
+    def fully_qualified_service(self):
+        """return the service name prefixed with package"""
+        return f"{self.package}.{self.service}" if self.package else self.service
+
+
+def parse_method_name(method_name: str) -> tuple[MethodName, bool]:
+    """
+    Infers the grpc service and method name from the handler_call_details.
+    e.g. /package.ServiceName/MethodName
+    """
+    if len(method_name.split("/")) < 3:
+        return MethodName(), False
+    _, package_service, method = method_name.split("/")
```

Comment:
Does this work if there are 4 parts to `split`?

```suggestion
    _, package_service, method = method_name.split("/")
```

Reply:
method name are guaranteed to have 3 split

Reply:
What about `package.service/Method/With/Slashes`? Or is that explicitly disallowed?

Reply:
that is not allowed.

Reply:
https://grpc.github.io/grpc/core/md_doc_internationalization.html <- it seems like that's not the case?